File: pkg/vm-low-orbit/tests/fixtures/guest/fifo_mod_1.go
//go:build fifo_mod_1

package main

//lint:file-ignore U1000 compiled file

import (
	"github.com/taubyte/go-sdk/i2mv/fifo"
)

//export fifo_1
func fifo_1() uint32 {
	data := []byte("hello world")
	id, ff := fifo.New(true)

	ff.Write(data)

	return id
}

File: pkg/vm-low-orbit/tests/fixtures/rust/srcs/fifo_mod_1.rs
extern crate taubyte_sdk;

use std::io::Write;

use taubyte_sdk::i2mv::fifo::WriteCloser;

#[no_mangle]
pub fn fifo_1() -> u32 {
    let mut ff = WriteCloser::new(true);
    ff.write("hello world".as_bytes()).unwrap();

    ff.id
}
